doc/design/switch.md

```diff
@@ -0,0 +1,31 @@
+### Design Doc: Switch
+
+### Background
+
+Many programming languages provide `switch` as a generalization of `if-elif-else`.  We want to add it to Fluid.
+
+The following example shows the usage of `fluid.switch`.
+
+```python
+a = fluid.Var(10)
+b = fluid.Var(0)
+
+cond = fluid.switch()
+with cond.case(fluid.less_equal(a, 10)):
+    fluid.print("Case 1")
+with cond.case(fluid.larger(a, 0)):
+    fluid.print("Case 2")
+with cond.default():
+    fluid.print("Case 3")
+```
```

Comment:
We need to scope `cond`s to underline that these three conds are in the same scope.

Currently, the cases should be adjacent, but the `with` syntax does not have that constraint semantically, so user tends to feel weird and write some wrong code.

what if user write some code like

```python
cond = fluid.switch()
with cond.case(case1):
    pass
with cond.case(case2):
    pass
b = a + c
with cond.case(cond3):
    pass
```
the `b = a + c` in the cases will break the global scope/continuous relations of a `switch`-cases.

Discussed with @jacquesqiao 

In lisp, a `cond` op is like
```lisp
;; cond has a scope to wrap all the following cases
(cond ((> a 2) "gt")
      ((= a 2) "eq")
      (t "lt")
) ;; cond
```
it seems better to write code as follows
```python
cond = fluid.switch()
with fluid.switch() as switch:
    '''
    A limitation inside a `switch`-block that no operators can insert between `case`s.  
    This rule is easier to check.
    '''
    with switch.case(fluid.less_equal(a, 10)):
           pass
    with switch.case(fluid.larger(a, 1)):
           pass
    switch()
```

@wangkuiyi @jacquesqiao

Reply:
maybe need a cond.scope() as follow, one cond must and only has one scope.
```python
cond = fluid.switch()
with cond.scope():
    '''
    A limitation inside a `switch`-block that no operators can insert between `case`s.  
    This rule is easier to check.
    '''
    with switch.case(fluid.less_equal(a, 10)):
           pass
    with switch.case(fluid.larger(a, 1)):
           pass
```